django/db/models/sql/compiler.py

```diff
@@ -453,6 +453,10 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     if self.connection.get_autocommit():
                         raise TransactionManagementError('select_for_update cannot be used outside of a transaction.')
 
+                    if with_limits and not self.connection.features.supports_select_for_update_with_limit:
+                        raise DatabaseError(
+                            'LIMIT/OFFSET not allowed with select_for_update on this database backend.'
+                        )
```

Comment:
I added this check to avoid potential bugs in 3rd party database backends, but I'm not convinced that it's necessary.